dali/python/nvidia/dali/shared_mem.py

```diff
@@ -0,0 +1,108 @@
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nvidia.dali import backend as _b
+
+class SharedMem:
+    """SharedMem allows you to allocate and access shared memory.
+    Provides memory view of the allocated memory via buf property.
+    You can transfer access to the same shared memory chunk by sending related
+    file descriptor available as fd property. Use SharedMem.allocate
+    to allocate new chunk of shared memory and SharedMem.open if
+    you received file descriptor to already existing memory chunk.
+
+    There is out of the box support for shared memory starting from Python3.8, though
+    the only way there to transfer the memory to other processes is via filename,
+    which might 'leak' if process was closed abruptly.
+
+    Parameters
+    ----------
+    `fd` : int
+        File descriptor identifying related shared memory object. Pass -1 to allocate new memory chunk.
```

Comment:
```suggestion
    `handle` : int
        A handle to the shared memory object. Pass None to allocate a new memory chunk.
```
1. Use non-OS-specific names
2. Translate None into OS-specific invalid handle value (like -1) inside the function

Reply:
Done